shards/shards.go

```diff
@@ -1064,7 +1065,10 @@ func (s *shardedSearcher) getLoaded() loaded {
 
 func mkRankedShard(s zoekt.Searcher) *rankedShard {
 	q := query.Const{Value: true}
-	result, err := s.List(context.Background(), &q, nil)
+	// We need to use UnsafeCtx here, otherwise we cannot return a proper
```

Comment:
Would there be a way to bypass `List` and load the repo metadata directly? So this just obviously becomes a low-level, privileged operation. This would introducing `systemtenant` just for this single privileged call. Or maybe it's not possible to refactor things to support this 🤔 

I'm not an expert on this code and totally lack context, so just ignore me if this not helpful :)

Reply:
My initial thoughts was around this, but I couldn't think of a nice way. EG we want to take into account tombstones/etc and anything else we may add to List in the future. So using List feels like the cleanest approach.